static/js/components/email/EmailCompositionDialog.js

```diff
@@ -28,25 +31,49 @@ type EmailDialogProps = {
   email:            EmailState,
   sendEmail:        () => void,
   title?:           string,
-  children?:        any,
+  subheadingType?:  string
+};
+
+const renderSimpleSubheading = (subheading: string) => (
+  <div className="subheading-section">
+    <h5 className="subheading">{ subheading }</h5>
+  </div>
+);
+
+const renderRoundedSubheading = (subheading: string) => (
+  <div className="subheading-section">
+    <Grid noSpacing={true}>
+      <Cell col={1} align={"middle"} className="subheading-recipient">TO:</Cell>
+      <Cell col={11}><h5 className="subheading rounded">{ subheading }</h5></Cell>
+    </Grid>
+  </div>
+);
+
+const emailTypeRendererMap = {
+  [SEARCH_EMAIL_TYPE]: renderSimpleSubheading,
+  [COURSE_EMAIL_TYPE]: renderRoundedSubheading
+};
```

Comment:
> if you call .get with a key the map in question does not contain it returns undefined

an Object does the same thing. `emailTypeRendererMap['nonsense']` == `undefined`. i jumped through these hoops mostly b/c of flow typing. `emailTypeRendererMap[renderType]` fails because `renderType` can be `undefined`. i want to get the default function if the key doesn't exist _or_ the key is null/undefined. addressing your comment above, i created for a couple reasons: (1) `getPropertyOrDefault` is a little bit clearer about intent than using `_.get` directly, and (2) there's a lot of functional overlap between ramda and lodash, so on the very very remote chance that we decided we wanted to get rid of lodash (or just import specific functions), we could change the implementation of that function. neither are terribly compelling reasons. i think it's likely for a developer to need to check docs whether they see this function or `_.get` with a third arg, and it's just a bit nicer to have that function documented locally, so i'm slightly leaning towards keeping it. let me know what you think